docs/storage.md

```diff
@@ -0,0 +1,57 @@
+# Storage components
+
+#### 1. Ledger
+The ledger is an ordered log of transactions with each transaction assigned a unique monotonically increasing positive integer called sequence number.
+Sequence numbers start from 1 and then increase by 1 for each new transaction. There are no gaps in sequence numbers. 
+The actual storage behind the ledger can be flat files with each transaction in a new line or a key value store like LevelDB with key being the sequence number and value being the transaction.
+A transaction is serialised (currently as MsgPack) before storing in the ledger, more on this in the Serialisation doc.
+Each node hosts several ledgers each identified by a unique id:
+-   Pool Ledger (id is 0): Contains transactions related to pool membership, like joining of new nodes, suspension of existing nodes, changing the IP/port or keys of existing nodes.
+-   Domain Ledger (id is 1): Contains transactions related to the core application logic. Currently it contains NYM transactions. The `indy-node` codebase extends this ledger with other identity transactions.
+-   Config Ledger (id is 2): Contains transactions related to the configuration parameters for which the pool needs to agree, eg. if each node of the pool needs to use the value `5` for a config variable `x`, then this ledger should contain transaction specifying the value of `x` as `5`. The `indy-node` codebase extends this ledger with source code update transactions.
+
+More ledgers can added by plugins. Each correct node should have exactly the same transactions for each ledger id.
+
+A ledger is associated with a [compact merkle tree](https://github.com/google/certificate-transparency/blob/master/python/ct/crypto/merkle.py). 
+Each new transaction is added to the ledger (log) and is also hashed (sha256) and this hash becomes a new leaf of the merkle tree which also 
```

Comment:
Please make each statement as a separate bullet point.

Reply:
I have broken the text into paragraphs.